server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBranch.tsx

```diff
@@ -73,6 +74,11 @@ export default class ComponentNavBranch extends React.PureComponent<Props, State
     this.mounted = false;
   }
 
+  differentBranches(a: Branch, b: Branch) {
+    // if main branch changes name, we should not close the dropdown
+    return a.isMain && b.isMain ? false : a.name !== b.name;
+  }
+
   handleClick = (event: React.SyntheticEvent<HTMLElement>) => {
```

Comment:
Nothing to do with your PR but I think clicking again on the link when the dropdown is open should close it.